cmd/sealer/cmd/utils/cluster_test.go

```diff
@@ -0,0 +1,203 @@
+// Copyright © 2021 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"net"
+	"testing"
+
+	"github.com/sealerio/sealer/cmd/sealer/cmd/types"
+	"github.com/sealerio/sealer/pkg/clusterfile"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_returnFilteredIPList(t *testing.T) {
+	tests := []struct {
+		name              string
+		clusterIPList     []net.IP
+		toBeDeletedIPList []net.IP
+		IPListExpected    []net.IP
+		wantErr           bool
+	}{
+		{
+			"test",
+			[]net.IP{net.ParseIP("10.10.10.1"), net.ParseIP("10.10.10.2"), net.ParseIP("10.10.10.3"), net.ParseIP("10.10.10.4")},
+			[]net.IP{net.ParseIP("10.10.10.1"), net.ParseIP("10.10.10.2"), net.ParseIP("10.10.10.3"), net.ParseIP("10.10.10.4")},
+			[]net.IP{},
+			false,
+		},
+		{
+			"test1",
+			[]net.IP{net.ParseIP("10.10.10.1"), net.ParseIP("10.10.10.2"), net.ParseIP("10.10.10.3"), net.ParseIP("10.10.10.4")},
+			[]net.IP{},
+			[]net.IP{net.ParseIP("10.10.10.1"), net.ParseIP("10.10.10.2"), net.ParseIP("10.10.10.3"), net.ParseIP("10.10.10.4")},
+			false,
+		},
+		{
+			"test2",
+			[]net.IP{net.ParseIP("10.10.10.1"), net.ParseIP("10.10.10.2"), net.ParseIP("10.10.10.3"), net.ParseIP("10.10.10.4")},
+			[]net.IP{net.ParseIP("10.10.10.4")},
+			[]net.IP{net.ParseIP("10.10.10.1"), net.ParseIP("10.10.10.2"), net.ParseIP("10.10.10.3")},
+			false,
+		},
+		{
+			"test3",
+			[]net.IP{},
+			[]net.IP{net.ParseIP("10.10.10.1"), net.ParseIP("10.10.10.2"), net.ParseIP("10.10.10.3"), net.ParseIP("10.10.10.4")},
+			[]net.IP{},
+			false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
It's not simple enough. Let's finish the last change.
IMO, "(res != nil) != tt.wantErr" means `if res != nil`. In this case, `wantErr` should be removed.
And let's do like this:
1. remove wantErr
2. 
```
res := removeIPList(tt.clusterIPList, tt.toBeDeletedIPList)
if res == nil {
continue
}

assert.Equal(t, tt.IPListExpected, res)
```
Or make the expected IPList to be `nil`
Then you don't have to make sure the `res != nil`.

Reply:
This judgment is inside a for loop, so you can't use continue?



>Or make the expected IPList to be nil

>Then you don't have to make sure the res ! = nil.

The meaning of this sentence is not quite understood

But my understanding is this:

If expected IPList is empty, an error will be reported when the res and expected IPList functions are detected